Apache-Beam-Jobs.md

```diff
@@ -227,158 +215,386 @@ error_pcoll = (
 
 `Runner`s provide the `run()` method used to visit every node (`PValue`) in the pipeline's DAG by executing the edges (`PTransform`s) to compute their values.  At Oppia, we use `DataflowRunner` to have our `Pipeline`s run on the [Google Cloud Dataflow service](https://cloud.google.com/dataflow).
 
-## Case studies
+## Writing Apache Beam Jobs
 
-The case studies are sorted in order of increasing complexity. Study the one that best suits your needs.
+For this section, we'll walk through the steps of implementing a job by writing one: `CountExplorationStatesJob`.
 
-If none of them help you implement your job, you may request a new one by adding a comment to [#13190](https://github.com/oppia/oppia/issues/13190) with answers to the following questions:
+It's helpful to begin by sketching a diagram of what you want the job to do. We recommend using pen and paper or a whiteboard, but in this wiki page we'll use ASCII art to keep the document self-contained.
 
-* Why do I want a new case study?
-* Why are the current case studies insufficient?
-* What answers would the "perfect" case study provide?
+Here's a diagram for the `CountExplorationStatesJob`:
 
-Then we'll start write a new Case Study to help you, and future contributors, as soon as we can (@brianrodri will always notify you of how long it'll take).
+    .--------------. Count states .--------. Sum .-------.
+    | Explorations | -----------> | Counts | --> | Total |
+    '--------------'              '--------'     '-------'
 
-### Case study: `CountAllModelsJob`
+> **TIP**: As illustrated, you don't need to know what the names of the `PTransform`s (edges) used in a diagram are. It's easy to look up the appropriate `PTransform` after drawing the diagram.
 
-**Difficulty:** Trivial
+Now that we have our bearings, let's get started on implementing the job.
 
-**Key Concepts:**
+### 1. Subclass the `base_jobs.JobBase` class and override the `run()` method
```

Comment:
Can you update the table of contents to include this step? We've included other level-3 headings in the TOC

Reply:
Done.